pkg/standalone-utils/contracts/LPOracleBase.sol

```diff
@@ -170,7 +170,14 @@ abstract contract LPOracleBase is ILPOracleBase, AggregatorV3Interface {
     }
 
     /// @inheritdoc ILPOracleBase
-    function calculateTVL(int256[] memory) public view virtual returns (uint256);
+    function computeTVL() public view returns (uint256) {
+        (int256[] memory prices, , ) = getFeedData();
+
+        return _computeTVL(prices);
+    }
+
+    /// @inheritdoc ILPOracleBase
+    function computeTVLGivenPrices(int256[] memory prices) public view virtual returns (uint256);
```

Comment:
I think we can just implement the function at the base layer; the current implementation in weighted / stable oracle contracts is exactly the same. We only need to override the internal `_computeTVL` in each oracle specialization.